src/hotspot/os/linux/osContainer_linux.cpp

```diff
@@ -60,11 +61,18 @@ void OSContainer::init() {
   if (cgroup_subsystem == NULL) {
     return; // Required subsystem files not found or other error
   }
-  // We need to update the amount of physical memory now that
-  // cgroup subsystem files have been processed.
-  if ((mem_limit = cgroup_subsystem->memory_limit_in_bytes()) > 0) {
+  if ((mem_limit = cgroup_subsystem->memory_limit_in_bytes()) > 0 &&
+       mem_limit < host_memory) {
```

Comment:
I really would like it if these two conditions were nested, that would be easier to understand.
```
if (memlimit valid and > 0) {
  if (memlimit <= phys) {
    phys = memlimit
  } else {
    log
  }

}

Reply:
Thank you; Addressed in `7f5307d69bb` (although possibly moot, once we resolve your other review comment https://github.com/openjdk/jdk/pull/9880#discussion_r947908655).

Presently, we trace log if the memory limit matches physical RAM. We could stop doing that: it doesn't seem too valuable. WDYT?

Reply:
Yes, its not that important.